api/osb/reference_instance_plugin.go

```diff
@@ -0,0 +1,363 @@
+package osb
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/Peripli/service-manager/constant"
+	"github.com/Peripli/service-manager/pkg/log"
+	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/pkg/web"
+	"github.com/Peripli/service-manager/storage"
+	"github.com/tidwall/gjson"
+	"net/http"
+)
+
+const ReferenceInstancePluginName = "ReferenceInstancePlugin"
+const planIDProperty = "plan_id"
+const catalogIDProperty = "catalog_id"
+const contextKey = "context"
+
+type referenceInstancePlugin struct {
+	repository       storage.TransactionalRepository
+	tenantIdentifier string
+}
+
+type osbObject = map[string]interface{}
+
+// NewCheckPlatformIDPlugin creates new plugin that checks the platform_id of the instance
+func NewReferenceInstancePlugin(repository storage.TransactionalRepository, tenantIdentifier string) *referenceInstancePlugin {
+	return &referenceInstancePlugin{
+		repository:       repository,
+		tenantIdentifier: tenantIdentifier,
+	}
+}
+
+// Name returns the name of the plugin
+func (p *referenceInstancePlugin) Name() string {
+	return ReferenceInstancePluginName
+}
+
+func (p *referenceInstancePlugin) Provision(req *web.Request, next web.Handler) (*web.Response, error) {
```

Comment:
let use a more meaningful name for the pointer caller

Reply:
referencePlugin